src/libraries/System.Private.Xml/src/System/Xml/Core/XmlReader.cs

```diff
@@ -534,7 +534,7 @@ public virtual object ReadElementContentAs(Type returnType, IXmlNamespaceResolve
                 return value;
             }
 
-            return returnType == typeof(string) ? string.Empty : XmlUntypedStringConverter.Instance.FromString(string.Empty, returnType, namespaceResolver);
+            return returnType == typeof(string) ? string.Empty : XmlUntypedConverter.Untyped.ChangeType(string.Empty, returnType, namespaceResolver);
```

Comment:
I believe the new implementation will not work with list types correctly. I think the correct solution might be just changing `Untyped` into `UntypedList` but you will need to add tests which cover this.

Reply:
https://github.com/dotnet/runtime/blob/aa2af309a12d31dd68de9dc3646d77b6f25d9d40/src/libraries/System.Private.Xml/src/System/Xml/Core/XmlReaderAsync.cs#L61